test/jasmine/tests/calcdata_test.js

```diff
@@ -236,6 +232,14 @@ describe('calculated data and points', function() {
                 expect(gd.calcdata[0][2]).toEqual(jasmine.objectContaining({x: 6, y: 12}));
                 expect(gd.calcdata[0][3]).toEqual(jasmine.objectContaining({x: 1, y: 13}));
                 expect(gd.calcdata[0][4]).toEqual(jasmine.objectContaining({x: 4, y: 14}));
+
+                // The auto-range feature currently eliminates unused category ticks on the left/right axis tails.
+                // The below test case reifies this current behavior, and checks proper order of categories kept.
+
+                var domTickTexts = Array.prototype.slice.call(document.querySelectorAll('g.xtick'))
+                    .map(function(e) {return e.__data__.text;});
+
+                expect(domTickTexts).toEqual(['b', 'x', 'a', 'd', 'z', 'e', 'c']);  // y, q and k has no data points
```

Comment:
@etpinard a few questions regarding this type of test:

1. I've used library-free querying of the DOM; I think it's good but let me know if I should import d3 for this purpose, or you could point out another test file as a pattern to follow.
2. I haven't yet bumped into axes/ticks tests of this nature, where we query the DOM, which is downstream of the 'viewModel' testing this file mostly has but upstream of image based testing. Do we need such DOM based testing, and are there preexisting DOM tests I should instead use that ensure e.g. proper category or tracing order?
3. If we need such tests, should I add similar ones for the other test cases in this file?